frontend/src/components/module-editor/LessonOptions.tsx

```diff
@@ -0,0 +1,74 @@
+    setFocus: (index: number) => void,
+}
+
+const LessonOptions = ({ text='', isFocused, isHovered=({hoveredState: false, index: 0}), index, setIsHovered, setFocus }: OptionsProps) => ( 
+  <>
+    {!isFocused ? (
+      <Button
+        key={index}
+        onClick={() => setFocus(index)}
+        variant="unstyled"
+        textAlign="left"
+        pl="10px"
+        minH="55px"
+        w="100%"
+        onMouseEnter={() => setIsHovered({hoveredState: true, index})}
+        onMouseLeave={() => setIsHovered({hoveredState: false, index})}
+      >
+        <Flex align='center' justify="space-between" pr="8px">
+          <Flex align='center'>
+            <IconButton visibility={(isHovered.hoveredState && isHovered.index === index) ? 'visible' : 'hidden'} aria-label='Drag Lesson' variant="unstyled" size="xs" icon={<DragHandleIconSvg/>} />
+            <p style={{marginLeft: "10px"}}> {text} </p>
+          </Flex>
+          <Flex align='center' pb='5px'>
+            <IconButton visibility={(isHovered.hoveredState && isHovered.index === index) ? 'visible' : 'hidden'} aria-label='Edit Lesson' variant="unstyled" fontSize="18px" size="sm" icon={<EditIcon />} />
+            <IconButton visibility={(isHovered.hoveredState && isHovered.index === index) ? 'visible' : 'hidden'} aria-label='Delete Lesson' variant="unstyled" fontSize="18px" size="sm" icon={<DeleteIcon />} />
+          </Flex>
+        </Flex>
+      </Button>
+    ) : (
+        <Button
+          key={index}
+          onClick={() => setFocus(index)}
+          variant="unstyled"
+          borderLeftColor="brand.royal"
+          borderLeftWidth="5px"
+          borderRadius="0"
+          bg="background.light"
+          textAlign="left"
+          pl="6px"
+          minH="55px"
+          w="100%"
+          onMouseEnter={() => setIsHovered({hoveredState: true, index})}
+          onMouseLeave={() => setIsHovered({hoveredState: false, index})}
+        >
+          <Flex align='center' justify="space-between" pr="8px">
+            <Flex align='center'>
+              <IconButton visibility={(isHovered.hoveredState && isHovered.index === index) ? 'visible' : 'hidden'} aria-label='Drag Lesson' variant="unstyled" size="xs" icon={<DragHandleIconSvg/>} />
+              <p style={{marginLeft: "10px"}}>
+                {text} 
+              </p>
+            </Flex>
+            <Flex align='center' pb='5px'>
+              <IconButton visibility={(isHovered.hoveredState && isHovered.index === index) ? 'visible' : 'hidden'} aria-label='Edit Lesson' variant="unstyled" fontSize="18px" size="sm" icon={<EditIcon />} />
+              <IconButton visibility={(isHovered.hoveredState && isHovered.index === index) ? 'visible' : 'hidden'} aria-label='Delete Lesson' variant="unstyled" fontSize="18px" size="sm" icon={<DeleteIcon />} />
+            </Flex>
+          </Flex>
+        </Button>
+    )}
+  </>
+);
+
+  
+export default LessonOptions;
```

Comment:
With the understanding that this component encompasses the whole lesson item, I think it'd be better to call it LessonItem instead of LessonOptions. I think LessonOptions would be more appropriate if we only focused on the options buttons showing up, but it seems like you are able to click on the lesson itself.